test/web_server_proxy_test.py

```diff
@@ -68,6 +68,43 @@ async def hello(request):
         yield DummyHttpServer(host=host, port=port, event=event, assertion_log=assertion_log)
 
 
+@contextlib.asynccontextmanager
+async def lifespan_ctx_manager(asgi_app):
+    state: Dict[str, Any] = {}
```

Comment:
Thinking about it, we should be able to refactor this to just use `LifespanManager` internally to make it a ~3 liner